### ISSUE TITLE
`psql` outputs it's `NOTICE` messages to stderr causing the action to fail

### ISSUE THREAD

**Issue by k-visscher** [Azure/postgresql#33]
`psql` outputs it's `NOTICE` messages to stderr causing the whole action to fail even if the exit code is `0`.
The offending lines of code:
https://github.com/Azure/postgresql/blob/f82d2b25075a9092d3ee987db9c77bcfc43fc6bc/src/PsqlFilesExecutor.ts#L27-L33
https://github.com/Azure/postgresql/blob/f82d2b25075a9092d3ee987db9c77bcfc43fc6bc/src/PsqlFilesExecutor.ts#L38-L40
`NOTICE`'s should be treated as warnings not errors those would be `EXCEPTION`'s.

Maybe there should be an option for choosing to ignore stderr and instead relying on the exit code of `psql`

It's trivial to get a `NOTICE` to fail the action.
Consider the following output generated by `dotnet ef migrations script -s Proj.Bootstrap -p Proj.Data --idempotent -o migration.sql`:
```SQL
CREATE TABLE IF NOT EXISTS "__EFMigrationsHistory" (
    "MigrationId" character varying(150) NOT NULL,
    "ProductVersion" character varying(32) NOT NULL,
    CONSTRAINT "PK___EFMigrationsHistory" PRIMARY KEY ("MigrationId")
);

START TRANSACTION;
....
COMMIT;
```

If the `__EFMigrationsHistory` table already exists, the action fails, because `psql:/migration.sql:5: NOTICE:  relation "__EFMigrationsHistory" already exists, skipping` gets written to stderr.

**Comment by keiguard** [comment]
As a hotfix around this, I added the following code to `lib/main.js`:
```js
catch (error) {
    if (error.message.includes('NOTICE:  relation "__EFMigrationsHistory" already exists, skipping')) {
        core.warning(`NOTICE:  relation "__EFMigrationsHistory" already exists, skipping`);
    }
    else {
        core.setFailed(`Error occurred while running action:\n${error}`);
    }
}
```

I could not figure out to get around a module missing error when compiling it, so I ended up using `ncc build` to compile all modules to one file. For anyone else that needs a hotfix, you can use my forked repo: 
(obviously be very careful using any repo that is not a trusted source):
`- uses: opt-pro/postgresql@v1.2.0`

**Comment by cotzo** [comment]
any updates on this issue? this is blocking any attempt to use this action for EntityFramework migrations

**Comment by MagnusSandgren** [comment]
I am experiencing the same issue. @k-visscher / @cotzo - what did you guys do as a workaround? 🙂

**Comment by MagnusSandgren** [comment]
Ah well, that's mildly terrifying 😅 I would like the workflow to fail if the migration fails. Thanks for the sesponse @cotzo 🙂

**Comment by k-visscher** [comment]
> I am experiencing the same issue. @k-visscher / @cotzo - what did you guys do as a workaround? 🙂

I went the `ontinue-on-error: true` YOLO route, that @cotzo also suggested, the project was small enough that I could afford to manually check every now and then if the step actually succeeded or not.

**Comment by MagnusSandgren** [comment]
After further inspection of the code, it looks to me like the entire [if block in PsqlFilesExecutor.ts](https://github.com/Azure/postgresql/blob/f82d2b25075a9092d3ee987db9c77bcfc43fc6bc/src/PsqlFilesExecutor.ts#L38-L40) can be, and should be removed. I believe this to be the case because the exec option [IgnoreReturnCode](https://github.com/actions/toolkit/blob/main/packages/exec/src/interfaces.ts#L28) is false by default and not overwritten by this repo. Meaning if the psql command returns with a return code other than 0, the exec package will [create](https://github.com/actions/toolkit/blob/c5035362ab300a9e43fcc7447d43e74be85c574a/packages/exec/src/toolrunner.ts#L667-L670), and [reject the promise](https://github.com/actions/toolkit/blob/c5035362ab300a9e43fcc7447d43e74be85c574a/packages/exec/src/toolrunner.ts#L536-L537) with the error. 

In other words - the if block will **only trigger** if the command writes to stderr and returns 0. However, if the command returns something other than 0, the error will flow right past the if block.

**Comment by seryckd** [comment]
Same problem here, NOTICES causing the action to fail.  It would be preferred that only the exit code should be checked.

The workaround I am using is to disable writing notices by adding this to the top of the SQL file.

SET client_min_messages TO WARNING;

It's not ideal, and I may need to use the workaround provided by @MagnusSandgren.

**Comment by mehradn7** [comment]
Hi @BALAGA-GAYATRI @aksm-ms @microsoftopensource @N-Usha @AmrutaKawade @kanika1894, can we have a status about this issue and more generally about this repository? Is it still being maintained? There are some pending issues such as this one, and the last commit dates back to 1 year ago so it would be great to know if we should wait for a fix or consider other alternatives.

Thanks!
